modules/core/src/lib-exports.js

```diff
@@ -0,0 +1,6 @@
+// eslint-disable-next-line
+if (process.env.NODE_ENV === 'production') {
+  module.exports = require('./lib.prod');
```

Comment:
The conventional wisdowm is that `require` kills tree shaking as that works on static analysis of `export`/`import`... 

Recommend validating that this actually leads to smaller apps. My fear is that this will trigger unconditional bundling of `lib.prod`.

Reply:
`core/lib` is almost entirely dependencies of the `Deck` class (with the exception of `LayerExtension` which is super light).

This change shaves about 1K off the minified bundle. However, the main concern of including the logging functions is performance, because even when `log.log` returns `noOp` we spend CPU cycles building the dynamic message strings.

Reply:
> core/lib is almost entirely dependencies of the Deck class (with the exception of LayerExtension which is super light).

You are saying these should always be bundled anyway? Fine, though
1. I recommend calling this out in comments above the require statements (E.g. `// 'require' defeats tree shaking but we always bundle these symbols anyway`).. 
2) When I do this type of setups, they tends to come back and bite me as the library evolves...

> However, the main concern of including the logging functions is performance, because even when log.log returns noOp we spend CPU cycles building the dynamic message strings.

Got it. FWIW, to handle this, probe.gl has always allowed for message strings to be supplied as functions that only gets evaluated when the logging actually is enabled, e.g instead of:

```
log.info(`Dynamic strings generated for ${variableName} every time we execute this`)();
```

```
log.info(() => `Don't want dynamic strings generated for ${variableName} when we are not actually logging`)();
```

In theory, this still mints a function (which should be way faster than building a string, but possibly not zero impact, depends on how the JS engine handles this).

Reply:
I ran some benchmark tests, it's actually 2-3x more expensive to create those functions inline.

Reply:
Thanks for checking and it only goes to prove that actual profiling beats assumptions every time. (Perhaps it is the captured variable that makes it expensive... but that doesn't matter).

In light of this, I'd consider modifying the various probe.gl log calls to take multiple string segment as parameters and then concat them inside the log functions (only when logging actually happens). That should create no small objects and simple parameter passing is typically extremely fast in modern JS engines.

I opened an issue in probe.gl: https://github.com/uber-web/probe.gl/issues/108. 

@Pessimistress Happy to help out with this one if this would be considered helpful to the ongoing optimization push.